bin/proof-generator/include/nil/proof-generator/prover.hpp

```diff
@@ -195,6 +201,20 @@ namespace nil {
                     BOOST_LOG_TRIVIAL(info) << "Proof written";
                 }
 
+                BOOST_LOG_TRIVIAL(info) << "Writing json proof to " << json_file_;
+                std::ofstream output_file;
+                output_file.open(json_file_);
```

Comment:
use `open_file(json_file_, std::ios::out)` (declared [here](https://github.com/NilFoundation/proof-producer/blob/4ddd85878d66f9b916e9fbdafe51180058e5bdaa/bin/proof-generator/include/nil/proof-generator/file_operations.hpp#L90C1-L99C10)), it checks if open was successfull